lgc/patch/ShaderInputs.cpp

```diff
@@ -723,3 +715,120 @@ ShaderInputs::ShaderInputUsage *ShaderInputs::getShaderInputUsage(ShaderStage st
     inputsUsage->inputs[inputKind] = std::make_unique<ShaderInputUsage>();
   return &*inputsUsage->inputs[inputKind];
 }
+
+// =====================================================================================================================
+// Try to optimize to use the accurate workgroupID arguments and set the function attribute for corresponding
+// amdgpu-no-workgroup-id-*. If partial components of WorkgroupID are used, the argument type will be aligned and the
+// relevant users will be re-targeted to the new generated lgc.shader.input.workgroupId.*. The related users of <3xi32>
+// workgroupId will be removed. We set amdgpu-no-workgroup-id-* function attribute to control the register tgid_*_en
+// later.
+//
+// @param pipelineState : Pipeline state
+// @param shaderStage : Shader stage
+// @param origFunc : The original entry point function
+void ShaderInputs::tryOptimizeWorkgroupId(PipelineState *pipelineState, ShaderStage shaderStage, Function *origFunc) {
+  assert(shaderStage == ShaderStageCompute);
+  bool useWorkgroupIds[3] = {false};
+  bool useWholeWorkgroupId = false;
+  SmallVector<Instruction *> extractVec3[3];
```

Comment:
Avoid redundant state by removing `useWorkgroupIds` and just checking `extractVec3[i].empty()` instead where it's needed.

Reply:
Fixed.